devito/types/dimension.py

```diff
@@ -1223,10 +1223,8 @@ def _arg_names(self):
             return ()
 
     def _arg_defaults(self, **kwargs):
-        # TODO: need a heuristic to pick a default incr size
-        # TODO: move default value to __new__
         try:
-            return {self.step.name: 8}
+            return {self.step.name: 16}
```

Comment:
nitpicking: Can we add a "TODO" to check the args for space order and have some empirical `2*space_order` or something?

Reply:
doing in a sec in upcoming PR, merging this one now